### PR TITLE
[Feature-5006][Worker] Add a configuration item to set whether the tenant is automatically created on Worker

### LINKED ISSUE #4995: [Bug][api] select an existing tenant, the workflow fails to run, indicating that the tenant does not exist
image
image

**Which version of Dolphin Scheduler:**
 -[dev]

### LINKED ISSUE #5006: [Feature][Worker] Add a configuration item to set whether the tenant is automatically created on Worker
**Describe the feature**

In the dev branch, the tenant executing a task will not be automatically created in worker. This will cause issue [#4995](https://github.com/apache/incubator-dolphinscheduler/issues/4995)

In the following two scenarios, it will cause inconvenience to users

- Product or operation people want to automatically create users instead of manually creating tenants
- In the docker container environment, the tenants cannot be created in advance. Once the container is pulled up again, all tenants will disappear

**Which version of Dolphin Scheduler:**

- [dev]

**Is your feature request related to a problem? Please describe.**

In the docker container environment, it's a very painful thing to create new tenants on every worker every time.

**Describe the solution you'd like**

Add a configuration item to set whether the tenant is automatically created on Worker

The default value of this configuration item is `false`, but it needs to be set to `true` in the container

The name of this configuration item is `worker.tenant.auto.create`

**Additional context**

Previous dev email discussion: https://lists.apache.org/thread.html/ra44b2e69759fcc980e4ed04c1811037bf0e743e47827fc2dcd1049d6%40%3Cdev.dolphinscheduler.apache.org%3E

### PR DESCRIPTION
**Pull request by chengshiwen**
## What is the purpose of the pull request

*[Feature][Worker] Add a configuration item to set whether the tenant is automatically created on Worker*

This closes #4995 and closes #5006

## Brief change log

*(for example:)*
  - *Add a configuration item named `worker.tenant.auto.create`*
  - *Add `createUserIfAbsent(String userName)` in `OSUtils.java`*

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

(and)

This change added tests and can be verified as follows:

  - *Manually verified the change by testing locally.*

### CODE REVIEW

**Comment by chengshiwen** [comment]
@CalvinKirs Please review

File: docker/build/conf/dolphinscheduler/worker.properties.tpl [hunk 1]
```diff
@@ -21,6 +21,9 @@ worker.exec.threads=${WORKER_EXEC_THREADS}
 # worker heartbeat interval
 worker.heartbeat.interval=${WORKER_HEARTBEAT_INTERVAL}
 
+# worker tenant auto create
+worker.tenant.auto.create=true
+
```
Comment (CalvinKirs) [inline docker/build/conf/dolphinscheduler/worker.properties.tpl:26]: The default value of false may be better. WDYT?

Comment (chengshiwen) [reply docker/build/conf/dolphinscheduler/worker.properties.tpl:26]: In `worker.properties`, the default value of `worker.tenant.auto.create` is `false`. But in the template file `docker/build/conf/dolphinscheduler/worker.properties.tpl` for container, set it to true will be better

Comment (chengshiwen) [reply docker/build/conf/dolphinscheduler/worker.properties.tpl:26]: In `worker.properties`
image

In `docker/build/conf/dolphinscheduler/worker.properties.tpl`
image

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**